### ISSUE TITLE
Can't start my application any more after editing model ('MapperWin637010291105362639, Version=1.1.36.1 not found)

### ISSUE THREAD

**Issue by vimarx** [eXpandFramework/eXpand#504]
I managed to open model editor and after changing the XMlcontent editor in MDO_DetailView and restart, the application fails to start with the

10.08.19 10:19:10.350	Resolve the 'DevExpress.Persistent.BaseImpl.v19.1' assembly
10.08.19 10:19:10.961	Customize TypesInfo
10.08.19 10:19:11.083	Generación de estructura de interfaz de usuario de aplicación...
10.08.19 10:19:21.658	ModelMapper assembly created
10.08.19 10:19:21.973	Resolve the 'MapperWin637010291105362639, Version=1.1.36.1, Culture=neutral, PublicKeyToken=null' assembly
vxerp.win.exe Error: 0 : 10.08.19 10:19:21.978	================================================================================
The error occurred:

	Type:       AggregateException
	Message:    Se han producido uno o varios errores.
	Data:       0 entries
	Stack trace:

   en System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   en System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   en DevExpress.ExpressApp.Win.WinApplication.SetupCore(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   en DevExpress.ExpressApp.XafApplication.Setup()
   en VxERP.Win.Program.Main(String[] arguments)
	----------------
	InnerException:

			Type:       ObjectCreatingException
			Message:    Unable to create an instance of the "ModelApplicationCreatorInfo" type.
Reason: "No se puede cargar el archivo o ensamblado 'MapperWin637010291105362639, Version=1.1.36.1, Culture=neutral, PublicKeyToken=null' ni una de sus dependencias. El sistema no puede encontrar el archivo especificado."
			Data:       0 entries
			Stack trace:

   en DevExpress.ExpressApp.Utils.Reflection.TypeData.CreateInstance(Object[] args)
   en DevExpress.ExpressApp.Model.Core.ModelApplicationCreator.CreateCreatorInfo(ModelApplicationCreator modelApplicationCreator)
   en DevExpress.ExpressApp.Model.Core.ModelApplicationCreator.get_CreatorInfo()
   en DevExpress.ExpressApp.Model.Core.ModelApplicationCreator.AddValidator[T](IModelNodeValidator`1 validator)
   en DevExpress.ExpressApp.SystemModule.SystemModule.AddModelNodeValidators(IModelNodeValidatorRegistrator validatorRegistrator)
   en DevExpress.ExpressApp.ApplicationModelManager.SetupModelApplicationCreator(ModelApplicationCreator creator, IEnumerable`1 modules, IEnumerable`1 aspects)
   en DevExpress.ExpressApp.ApplicationModelManager.Setup(ITypesInfo typesInfo, IEnumerable`1 boModelTypes, IEnumerable`1 modules, IEnumerable`1 controllers, IEnumerable`1 applicationLocalizerTypes, IEnumerable`1 applicationAspects, ModelStoreBase applicationModelDifferenceStore, String modelAssemblyFile)
   en DevExpress.ExpressApp.XafApplication.CreateModelManager(IEnumerable`1 boModelTypes)
   en DevExpress.ExpressApp.Win.WinApplication.CreateModelManager(IEnumerable`1 boModelTypes)
   en DevExpress.ExpressApp.XafApplication.Setup(ExpressApplicationSetupParameters parameters)
   en DevExpress.ExpressApp.XafApplication.SetupCore(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   en DevExpress.ExpressApp.Win.WinApplication.<>n__0(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   en DevExpress.ExpressApp.Win.WinApplication.<>c__DisplayClass77_0.<SetupCore>b__0()
   en System.Threading.Tasks.Task.InnerInvoke()
   en System.Threading.Tasks.Task.Execute()
			----------------
			InnerException:

					Type:       FileNotFoundException
					Message:    No se puede cargar el archivo o ensamblado 'MapperWin637010291105362639, Version=1.1.36.1, Culture=neutral, PublicKeyToken=null' ni una de sus dependencias. El sistema no puede encontrar el archivo especificado.
					Data:       0 entries
					Stack trace:

   en ModelApplicationCreatorInfo..ctor(ModelApplicationCreator applicationCreator)
					InnerException is null

--------------------------------------------------------------------------------

**Comment by vimarx** [comment]
had to delete modelassembly.dll

**Comment by vimarx** [comment]
and startup.profile

**Comment by apobekiaris** [comment]
hard to follow did u resolve it?

**Comment by vimarx** [comment]
btw I had a C:\Users\gmarzioni\AppData\Local\Temp\ModelMapperModule\vxerp path with each mapperwinxxxx.dll

**Comment by vimarx** [comment]
the logs says "10.08.19 13:58:30.588	Resolve the 'MapperWin637010420746958963, Version=1.1.36.1, Culture=neutral, PublicKeyToken=null' assembly"
but the Mapperwinxxxx.dll generated is a difrrent one

**Comment by vimarx** [comment]
If I delete the modelassembly. it start ok. but if I start it a second time (this second time the modelassembly is already there), it fails again, so it is not a workaround

**Comment by apobekiaris** [comment]
>MapperWin637010291105362639

latest version store the assemblies in temp, along with related application name. Whats the location of this file? 

though from the version we can tell that is generated from the latest modelmapper

**Comment by vimarx** [comment]
C:\Users\gmarzioni\AppData\Local\Temp\ModelMapperModule\vxerp

wich is temp+app.name (vxerp)

but the numbers are different

**Comment by vimarx** [comment]
Also now I found this exception on startup
Type:       AggregateException
	Message:    Se han producido uno o varios errores.
	Data:       0 entries
	Stack trace:

   en System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   en System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   en DevExpress.ExpressApp.Win.WinApplication.SetupCore(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   en DevExpress.ExpressApp.XafApplication.Setup()
   en VxERP.Win.Program.Main(String[] arguments)
	----------------
	InnerException:

			Type:       FileNotFoundException
			Message:    El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
			Data:       0 entries
			Stack trace:

   en System.Reflection.RuntimeAssembly.nLoadFile(String path, Evidence evidence)
   en System.Reflection.Assembly.LoadFile(String path)
   en Xpand.XAF.Modules.ModelMapper.Services.TypeMapping.TypeMappingService.<>c__DisplayClass90_0.<Init>b__10(Boolean _)
   en System.Reactive.Linq.ObservableImpl.Select`2.Selector._.OnNext(TSource value)
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Reactive.PlatformServices.ExceptionServicesImpl.Rethrow(Exception exception)
   en System.Reactive.Stubs.<>c.<.cctor>b__2_1(Exception ex)
   en System.Reactive.AnonymousSafeObserver`1.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.HalfSerializer.ForwardOnError[T](ISink`1 sink, Exception ex, Int32& wip, Exception& error)
   en System.Reactive.Linq.ObservableImpl.TakeUntil`2._.OnError(Exception ex)
   en System.Reactive.Linq.ObservableImpl.Merge`1.Observables._.InnerObserver.OnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Linq.ObservableImpl.Switch`1._.InnerObserver.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Linq.ObservableImpl.SelectMany`2.ObservableSelector._.InnerObserver.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Subjects.FastImmediateObserver`1.EnsureActive(Int32 count)
   en System.Reactive.Subjects.FastImmediateObserver`1.EnsureActive()
   en System.Reactive.Subjects.ReplaySubject`1.ReplayBase.OnError(Exception error)
   en System.Reactive.Subjects.ReplaySubject`1.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Linq.ObservableImpl.SelectMany`2.ObservableSelector._.InnerObserver.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Linq.ObservableImpl.Switch`1._.InnerObserver.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Subjects.FastImmediateObserver`1.EnsureActive(Int32 count)
   en System.Reactive.Subjects.FastImmediateObserver`1.EnsureActive()
   en System.Reactive.Subjects.ReplaySubject`1.ReplayBase.OnError(Exception error)
   en System.Reactive.Subjects.ReplaySubject`1.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Subjects.Subject`1.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Sink`2.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Linq.ObservableImpl.Switch`1._.OnError(Exception error)
   en System.Reactive.Sink`1.ForwardOnError(Exception error)
   en System.Reactive.Linq.ObservableImpl.Select`2.Selector._.OnNext(TSource value)
   en System.Reactive.Linq.ObservableImpl.All`1._.OnCompleted()
   en System.Reactive.Sink`1.ForwardOnCompleted()
   en System.Reactive.Linq.ObservableImpl.Do`1.Observer._.OnCompleted()
   en System.Reactive.Sink`1.ForwardOnCompleted()
   en System.Reactive.Sink`2.OnCompleted()
   en System.Reactive.Subjects.FastImmediateObserver`1.EnsureActive(Int32 count)
   en System.Reactive.Subjects.FastImmediateObserver`1.EnsureActive()
   en System.Reactive.Subjects.ReplaySubject`1.ReplayBase.OnCompleted()
   en System.Reactive.Subjects.ReplaySubject`1.OnCompleted()
   en Xpand.XAF.Modules.ModelMapper.Services.TypeMapping.TypeMappingService.Start()
   en System.Reactive.Linq.ObservableImpl.Finally`1._.Dispose(Boolean disposing)
   en System.Reactive.Sink`1.Dispose()
   en System.Reactive.Disposables.Disposable.TryDispose(IDisposable& fieldRef)
   en System.Reactive.Sink`1.Dispose(Boolean disposing)
   en System.Reactive.Sink`1.Dispose()
   en System.Reactive.Disposables.Disposable.TryDispose(IDisposable& fieldRef)
   en System.Reactive.Linq.ObservableImpl.Switch`1._.OnCompleted()
   en System.Reactive.Sink`1.ForwardOnCompleted()
   en System.Reactive.Sink`2.OnCompleted()
   en System.Reactive.Sink`1.ForwardOnCompleted()
   en System.Reactive.Sink`2.OnCompleted()
   en System.Reactive.Sink`1.ForwardOnCompleted()
   en System.Reactive.Linq.ObservableImpl.Switch`1._.OnCompleted()
   en System.Reactive.Sink`1.ForwardOnCompleted()
   en System.Reactive.Sink`2.OnCompleted()
   en System.Reactive.Sink`1.ForwardOnCompleted()
   en System.Reactive.Linq.ObservableImpl.ToObservableRecursive`1._.LoopRec(IScheduler scheduler)
   en System.Reactive.Linq.ObservableImpl.ToObservableRecursive`1._.<>c.<LoopRec>b__5_0(IScheduler innerScheduler, _ this)
   en System.Reactive.Concurrency.ScheduledItem`2.InvokeCore()
   en System.Reactive.Concurrency.CurrentThreadScheduler.Trampoline.Run(SchedulerQueue`1 queue)
   en System.Reactive.Concurrency.CurrentThreadScheduler.Schedule[TState](TState state, TimeSpan dueTime, Func`3 action)
   en System.Reactive.Concurrency.LocalScheduler.Schedule[TState](TState state, Func`3 action)
   en System.Reactive.Concurrency.Scheduler.ScheduleAction[TState](IScheduler scheduler, TState state, Action`1 action)
   en System.Reactive.Producer`2.SubscribeRaw(IObserver`1 observer, Boolean enableSafeguard)
   en System.ObservableExtensions.SubscribeSafe[T](IObservable`1 source, IObserver`1 observer)
   en System.Reactive.Linq.ObservableImpl.Switch`1._.OnNext(IObservable`1 value)
   en System.Reactive.Sink`1.ForwardOnNext(TTarget value)
   en System.Reactive.Linq.ObservableImpl.Select`2.Selector._.OnNext(TSource value)
   en System.Reactive.Sink`1.ForwardOnNext(TTarget value)
   en System.Reactive.Linq.ObservableImpl.Select`2.Selector._.OnNext(TSource value)
   en System.Reactive.Sink`1.ForwardOnNext(TTarget value)
   en System.Reactive.Linq.ObservableImpl.FirstAsync`1.Sequence._.OnNext(TSource value)
   en System.Reactive.Sink`1.ForwardOnNext(TTarget value)
   en System.Reactive.Linq.ObservableImpl.Switch`1._.InnerObserver.OnNext(TSource value)
   en System.Reactive.Subjects.Subject`1.OnNext(T value)
   en Xpand.XAF.Modules.Reactive.ReactiveModule.ExtendModelInterfaces(ModelInterfaceExtenders extenders)
   en DevExpress.ExpressApp.ApplicationModelManager.CreateModelApplicationCreatorProperties(IEnumerable`1 modules, IEnumerable`1 controllers, String assemblyFileAbsolutePath)
   en DevExpress.ExpressApp.ApplicationModelManager.Setup(ITypesInfo typesInfo, IEnumerable`1 boModelTypes, IEnumerable`1 modules, IEnumerable`1 controllers, IEnumerable`1 applicationLocalizerTypes, IEnumerable`1 applicationAspects, ModelStoreBase applicationModelDifferenceStore, String modelAssemblyFile)
   en DevExpress.ExpressApp.XafApplication.CreateModelManager(IEnumerable`1 boModelTypes)
   en DevExpress.ExpressApp.Win.WinApplication.CreateModelManager(IEnumerable`1 boModelTypes)
   en DevExpress.ExpressApp.XafApplication.Setup(ExpressApplicationSetupParameters parameters)
   en DevExpress.ExpressApp.XafApplication.SetupCore(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   en DevExpress.ExpressApp.Win.WinApplication.<>n__0(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   en DevExpress.ExpressApp.Win.WinApplication.<>c__DisplayClass77_0.<SetupCore>b__0()
   en System.Threading.Tasks.Task.InnerInvoke()
   en System.Threading.Tasks.Task.Execute()
			InnerException is null

**Comment by vimarx** [comment]
this is my temp path after several tries
![image](https://user-images.githubusercontent.com/226800/62824827-7fbc6200-bb79-11e9-91a5-836ea78f931a.png)

**Comment by vimarx** [comment]
the issue is very easy to reproduce (attached sample)

1) start an app as standalone (not in vs), modelassembly is creaded and starts Ok
2) start it a second time. it fails
[Varios.zip](https://github.com/eXpandFramework/eXpand/files/3489143/Varios.zip)

**Comment by apobekiaris** [comment]
great I let u know then once I rpro and fix it.

**Comment by vimarx** [comment]
my temp path after several tries with this sample app
![image](https://user-images.githubusercontent.com/226800/62824939-56500600-bb7a-11e9-8b5c-362003ae8054.png)
it may be generating garbage ....

**Comment by apobekiaris** [comment]
see the docs when assembly generation happens, this might be expected

**Comment by vimarx** [comment]
BTW I had xpand and dx in GAC

**Comment by isatahiri** [comment]
I'm having the same issue with eXpqnd 19.1.502.8 : 
The error occurred:

	Type:       AggregateException
	Message:    Une ou plusieurs erreurs se sont produites.
	Data:       0 entries
	Stack trace:

   à System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   à System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   à DevExpress.ExpressApp.Win.WinApplication.SetupCore(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   à DevExpress.ExpressApp.XafApplication.Setup()
   à PME_Affaire.Win.Program.Main() dans C:\Users\isata\Documents\PME Projects\PME_Affaire\PME_Affaire.Win\Program.cs:ligne 57
	----------------
	InnerException:

			Type:       ObjectCreatingException
			Message:    Unable to create an instance of the "ModelApplicationCreatorInfo" type.
Reason: "Impossible de charger le fichier ou l'assembly 'MapperWin637010632231313945, Version=1.1.36.1, Culture=neutral, PublicKeyToken=null' ou une de ses dépendances. Le fichier spécifié est introuvable."
			Data:       0 entries
			Stack trace:

   à DevExpress.ExpressApp.Utils.Reflection.TypeData.CreateInstance(Object[] args)
   à DevExpress.ExpressApp.Model.Core.ModelApplicationCreator.CreateCreatorInfo(ModelApplicationCreator modelApplicationCreator)
   à DevExpress.ExpressApp.Model.Core.ModelApplicationCreator.get_CreatorInfo()
   à DevExpress.ExpressApp.Model.Core.ModelApplicationCreator.AddValidator[T](IModelNodeValidator`1 validator)
   à DevExpress.ExpressApp.SystemModule.SystemModule.AddModelNodeValidators(IModelNodeValidatorRegistrator validatorRegistrator)
   à DevExpress.ExpressApp.ApplicationModelManager.SetupModelApplicationCreator(ModelApplicationCreator creator, IEnumerable`1 modules, IEnumerable`1 aspects)
   à DevExpress.ExpressApp.ApplicationModelManager.Setup(ITypesInfo typesInfo, IEnumerable`1 boModelTypes, IEnumerable`1 modules, IEnumerable`1 controllers, IEnumerable`1 applicationLocalizerTypes, IEnumerable`1 applicationAspects, ModelStoreBase applicationModelDifferenceStore, String modelAssemblyFile)
   à DevExpress.ExpressApp.XafApplication.CreateModelManager(IEnumerable`1 boModelTypes)
   à DevExpress.ExpressApp.Win.WinApplication.CreateModelManager(IEnumerable`1 boModelTypes)
   à DevExpress.ExpressApp.XafApplication.Setup(ExpressApplicationSetupParameters parameters)
   à DevExpress.ExpressApp.XafApplication.SetupCore(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   à DevExpress.ExpressApp.Win.WinApplication.<>n__0(String applicationName, IList`1 objectSpaceProviders, ApplicationModulesManager modulesManager, ISecurityStrategyBase security)
   à DevExpress.ExpressApp.Win.WinApplication.<>c__DisplayClass77_0.<SetupCore>b__0()
   à System.Threading.Tasks.Task.InnerInvoke()
   à System.Threading.Tasks.Task.Execute()
			----------------
			InnerException:

					Type:       FileNotFoundException
					Message:    Impossible de charger le fichier ou l'assembly 'MapperWin637010632231313945, Version=1.1.36.1, Culture=neutral, PublicKeyToken=null' ou une de ses dépendances. Le fichier spécifié est introuvable.
					Data:       0 entries
					Stack trace:

   à ModelApplicationCreatorInfo..ctor(ModelApplicationCreator applicationCreator)
					InnerException is null


I tried to delete the tmp folder but not change

**Comment by expand** [comment]
The [DevExpress.XAF](https://github.com/eXpandFramework/DevExpress.XAF) repository includes commits that relate to this task:

* [ModelMapper: wrong configuration hash leads to compilation on each run #504](https://github.com/eXpandFramework/DevExpress.XAF/commit/2c6a3d67c373f90c0d80c5e8b0330e97f6b5e5f2)

Please update the related Nuget packages and test if issues is addressed. These are nightly nuget packages available only from our [NugetServer](https://xpandnugetserver.azurewebsites.net/nuget/).

If you do not use the Xpand.XAF.Modules directly but through a module of the main eXpandFramework project, please wait for the bot to notify you again when integration is finished or update the related packages manually.

Thanks a lot for your contribution.

**Comment by vimarx** [comment]
works Ok!!
